src/peft/peft_model.py

```diff
@@ -1779,7 +1779,7 @@ def _cpt_forward(
             else:
                 N_tokens = input_ids.shape[1]
             input_type_mask = torch.zeros((batch_size, N_tokens)).to(device)
-            input_type_mask[:, -1] = 4
+            input_type_mask[:, :] = 4
```

Comment:
`input_type_mask.fill_(4)` would also work. Could you add a short comment on what "4" means here?

Reply:
4 is the id for the tokens used for the loss calculation. I changed the code to 
input_type_mask = torch.ones((batch_size, N_tokens)).to(device) * 4